enhancements/cluster-logging/loki-gateway-configuration.md

```diff
@@ -0,0 +1,517 @@
+
+* Static Mode - It will offer static OPA configuration with a simple default rego file that will be part of the lokiStack gateway deployment as a static file. This static rego file needs to understand the RbacSpec. This makes `RbacSpec` mandatory.
+
+* Dynamic Mode - It will offer static OPA configuration along with an OPA url to delegate authorization of OIDC extracted subject information to a third-party OPA-compatible endpoint. This makes `OPA URL` mandatory and ignores `RbacSpec`.
+
+* Openshift Logging Mode - It will not need any information from the user. The lokiStack gateway deployment will configure dynamic OPA authorization itself using [opa-openshift](https://github.com/observatorium/opa-openshift/pull/1), because the sidecar will handle authorization based on SubjectAccessReviews against the API server.
+
+### LokiStack CR configuration - _Static Mode_
+
+In this mode, user will need to provide tenants, oidc and rbac configuration madatorily in the LokiStack CR. The lokiStack gateway deployment will itself create a static rego file to understand the rbacSpec.
+
+The LokiStack CR will look like this:
+
+```yaml
+tenants:
+    mode: static
+    authentication:
+    - name: tenant-a
+      oidc:
+        issuerURL: https://127.0.0.1:5556/dex
+        redirectURL: https://localhost:8443/oidc/tenant-a/callback
+        usernameClaim: test
+        groupClaim: test
+    authorization:
+      rbac:
+        roleBindings:
+        - name: tenant-a
+          roles:
+          - read-write
+          subjects:
+          - kind: user
+            name: admin@example.com
+        roles:
+        - name: read-write
+          permissions:
+          - read
+          - write
+          resources:
+          - metrics
+          tenants:
+          - tenant-a
+```
+
+The LokiStack Gateway deployment will need to create a static rego file which will include:
+
+```rego
+package observatorium
+
+import input
+import data.roles
+import data.roleBindings
+
+default allow = false
+
+allow {
+  some roleNames
+  roleNames = roleBindings[matched_role_binding[_]].roles
+  roles[i].name == roleNames[_]
+  roles[i].resources[_] = input.resource
+  roles[i].permissions[_] = input.permission
+  roles[i].tenants[_] = input.tenant
+}
+
+matched_role_binding[i] {
+  roleBindings[i].subjects[_] == {"name": input.subject, "kind": "user"}
+}
+
+matched_role_binding[i] {
+  roleBindings[i].subjects[_] == {"name": input.groups[_], "kind": "group"}
+}
+```
+
+### LokiStack CR configuration - _Dynamic Mode_
+
+In this mode, user will need to configure tenants, oidc and an OPA URL madatorily in the LokiStack CR. This OPA url will delegate authorization of OIDC extracted subject information to a third-party OPA-compatible endpoint.
+
+During the reconcilation of LokiStack CR, the user provided details will be validated. If anything is missing or provided extra, the Loki operator will give out an error.
```

Comment:
We don't need to mention validation of required fields here. However we can leave a word that mixing static RBAC with dynamic mode results in a LokiStack CR degraded condition.